src/main/java/com/intershop/version/semantic/SemanticVersion.java

```diff
@@ -16,294 +16,81 @@
-                        if (matcher.find()) {
-                            incrementState = getIncrementState(matcher.group(1).toLowerCase());
-                            String someNumberStr = matcher.group(2);
-                            if (!JETTY_GA_PATTERN.matcher(parts[i]).find())
-                            {
-                                numbers[3] = Integer.parseInt(someNumberStr);
-                            }
-                        }
-                        else
-                        {
-                            if (i>3)
-                            {
-                                isIncrementable = false;
-                                break;
-                            }
-                            Integer value = getDigit(parts[i]);
-                            if (value == null)
-                            {
-                                isIncrementable = false;
-                                break;
-                            }
-                            if (foundIncrementStateAlone)
-                            {
-                                numbers[i - 1] = value;
-                            }
-                            else
-                            {
-                                numbers[i] = value;
-                            }
-                        }
-                    }
-                }
-            }
-        }
-        catch(NumberFormatException e)
-        {
-            isIncrementable = false;
-        }
-        major = numbers[0];
-        minor = numbers[1];
-        patch = numbers[2];
-        increment = numbers[3];
-        this.isSemantic = version.equals(major + "." + minor + "." + patch);
-    }
-
-    private static ReleaseType getIncrementState(String lowerCased)
-    {
-        if ("final".equals(lowerCased) || "ga".equals(lowerCased))
-        {
-            return ReleaseType.GA;
-        }
-        if ("rc".equals(lowerCased))
-        {
-            return ReleaseType.RC;
-        }
-        if ("dev".equals(lowerCased))
-        {
-            return ReleaseType.DEV;
-        }
-        if (JETTY_GA_PATTERN.matcher(lowerCased).find())
-        {
-            return ReleaseType.GA;
-        }
-        return null;
-    }
-
-    private static Integer getDigit(String part)
-    {
-        Integer value = part.matches("^\\d+$") ? Integer.parseInt(part) : null;
-        if (value != null && value < 1000)
-        {
-            return value;
-        }
-        return null;
-    }
-
-    /**
-     * @return {@code true} if the number schema allows sorting
-     */
-    public boolean isIncrementable()
-    {
-        return isIncrementable;
-    }
-
     /**
-     * @return {@code true} if the number follows the semantic version schema
+     * @param version version as string
+     * @return semantic version to retrieve semantic meaning
      */
-    public boolean isSemantic()
+    static SemanticVersion valueOf(String version)
```

Comment:
Having convenient method like this factory method might be of use but it also introduces a circular dependency between SemanticVersion and VersionResolver. Think about to remove this method.

Reply:
I introduced the method to be as much compatible as possible.